pkg/controller/controller.go

```diff
@@ -582,6 +594,28 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 			}
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.CrossNamespaceVolumeDataSource) {
+		if claim.Spec.DataSourceRef != nil {
```

Comment:
This logic looks the same as the DataSource logic. Can we create a common function that takes in Name and Kind?

Reply:
Fixed.